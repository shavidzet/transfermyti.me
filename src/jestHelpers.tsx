/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
