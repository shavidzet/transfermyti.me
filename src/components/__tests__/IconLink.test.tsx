import React from "react";
import { render, RenderResult } from "~/jestHelpers";
import IconLink from "../IconLink";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    iconName: "heart",
    color: "ruby",
    size: 24,
    ...propOverrides,
  };

  const wrapper = render(<IconLink {...props} />);

  return { props, wrapper };
};

describe("the <IconLink> component", () => {
  test("matches its snapshot with valid props", () => {
    const { wrapper } = setup();

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });

  test("uses a default size if props.size is undefined", () => {
    const { wrapper } = setup({ size: undefined });

    expect(wrapper.container.firstElementChild).toHaveStyleRule(
      "width",
      "24px",
    );
    expect(wrapper.container.firstElementChild).toHaveStyleRule(
      "height",
      "24px",
    );
  });
});
