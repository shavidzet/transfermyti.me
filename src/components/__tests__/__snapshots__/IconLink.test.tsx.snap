// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <IconLink> component matches its snapshot with valid props 1`] = `
.emotion-1 {
  display: block;
  overflow: hidden;
  white-space: nowrap;
  width: 24px;
  height: 24px;
}

.emotion-1:hover path {
  fill: #840F1C;
}

.emotion-0 {
  fill: rgb(165,19,35);
}

<a
  class="emotion-1"
  color="ruby"
>
  <svg
    aria-describedby="icon-heart-desc"
    aria-labelledby="icon-heart-title"
    height="24"
    role="presentation"
    version="1.1"
    viewBox="0 0 1792 1536"
    width="24"
  >
    <title
      id="icon-heart-title"
    >
      Heart Icon
    </title>
    <desc
      id="icon-heart-desc"
    >
      A filled-in heart (the cartoon kind, not one with ventricles).
    </desc>
    <path
      class="emotion-0"
      d="M896 1536q-26 0-44-18L228 916q-10-8-27.5-26 T145 824.5 77 727 23.5 606 0468q0-220 127-344T478 0 q62 0 126.5 21.5t120 58T820 148t76 68q36-36 76-68t95.5-68.5 120-58T1314 0q224 0 351 124t127 344 q0 221-229 450l-623600q-18 18-44 18z"
      data-testid="svg-icon-path"
    />
  </svg>
</a>
`;
